ffi/capi/src/units_converter.rs

```diff
@@ -69,13 +70,30 @@ pub mod ffi {
     pub struct ICU4XMeasureUnitParser<'a>(pub MeasureUnitParser<'a>);
 
     impl<'a> ICU4XMeasureUnitParser<'a> {
+        /// Parses the CLDR unit identifier (e.g. `meter-per-square-second`) and returns the corresponding [`ICU4XMeasureUnit`].
+        /// Returns an error if the unit identifier is not valid or the `unit_id` is not a valid UTF-8 string.
+        #[diplomat::rust_link(
+            icu::experimental::units::measureunit::MeasureUnitParser::parse,
+            FnInStruct
+        )]
+        pub fn parse_unit_identifier<'text>(
+            &self,
+            unit_id: &'text DiplomatStr,
+        ) -> Result<Box<ICU4XMeasureUnit>, ICU4XError> {
```

Comment:
As discussed:

- There should be one function, `parse_unit_identifier`, which takes a DiplomatStr
- It should call `try_from_bytes` in Rust